pkgs/applications/misc/web-media-controller/default.nix

```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchFromGitHub, cmake, pkgconfig, glib, pcre, json-glib }:
+
+stdenv.mkDerivation rec {
+  pname = "wmc-mpris";
+  name = "${pname}-unstable";
+  version = "2019-04-12";
+
+  src = fetchFromGitHub {
+    owner = "f1u77y";
+    repo = pname;
+    rev = "3b92847c576662732984ad791d6c7899a39f7787";
+    sha256 = "0q19z0zx53pd237x529rif21kliklwzjrdddx8jfr9hgghjv9giq";
+  };
+
+  nativeBuildInputs = [ cmake pkgconfig ];
+  buildInputs = [ glib pcre json-glib ];
+  preConfigure = ''
+    cmakeFlags="-DCHROMIUM_MANIFEST_DESTINATION=$out/etc/chromium/native-messaging-hosts \
```

Comment:
Why is this in `preConfigure`?
You can set `cmakeFlags` as an attribute.

Reply:
I'm glad you've noticed that and I can assure you I wished I could use the `cmakeFlags` as an attribute. The issue is, that `$out` is taken literally when used in this attribute and using `${out}` fails with the error that `out` is undefined. I've found an old archived mailing list thread stating (at least at 2013) that this was the only way to do it:

https://nixos.org/nix-dev/2013-May/011157.html

Reply:
That's because `${out}` is string interpolation.
It would work fine if you set it to the environment variable `$out` and just hope that the shell fixes that up.

But nix has a builtin `placeholder` that gives us the path to `out` so you can do
```
cmakeFlags = [
  "-DCHROMIUM_MANIFEST_DESTINATION=${placeholder ''out''}/..."
];

Reply:
> It would work fine if you set it to the environment variable $out and just hope that the shell fixes that up.

Exactly as experienced in that mailing list thread, the shell doesn't `eval`uate the value of this variable. However, your suggestion of using `placeholder` worked. Updating.